walletlibrary/src/main/java/com/microsoft/walletlibrary/networking/entities/openid4vci/credentialmetadata/CredentialMetadata.kt

```diff
@@ -27,4 +34,57 @@ internal data class CredentialMetadata(
 
     // Display information for the issuer.
     val display: List<LocalizedIssuerDisplayDefinition>? = null,
-)
\ No newline at end of file
+) {
+    fun transformLocalizedIssuerDisplayDefinitionToRequesterStyle(): RequesterStyle {
+        display?.forEach { displayDefinition ->
+            displayDefinition.locale?.let { language ->
+                ConfigurationCompat.getLocales(Resources.getSystem().configuration)
+                    .getFirstMatch(
+                        arrayOf(language)
+                    )?.let {
+                        return VerifiedIdManifestIssuerStyle(displayDefinition.name ?: "")
+                    }
+            }
+        }
+        return VerifiedIdManifestIssuerStyle(display?.first()?.name ?: "")
```

Comment:
Kotlin question, so this basically traverses through all of the displayDefinition, for displayDef in displayDefinitions, look at the locale, if it matches one of the preferred languages in the OS, use that displayDefinition.name?

What if the first locale of the first displayDefinition is french maybe and the locales in the ConfigurationCompat is [en, french], so the user prefers english over french but since the first display Definition is french it will return that one. 

Should we traverse through the preferred languages first to see if there are any displayDefinitions that are the user's prefered language first instead?

Reply:
Oh yes, good catch, preferred languages should be traversed first. I remember I made the change, maybe it is in the future PR transforming the models and completing the request. I will make the change here.